rpcs3/rpcs3qt/gs_frame.cpp

```diff
@@ -299,6 +304,70 @@ void gs_frame::flip(draw_context_t, bool /*skip_frame*/)
 	}
 }
 
+void gs_frame::take_screenshot(std::vector<u8> ss_data, const u32 ss_width, const u32 ss_height)
+{
+	std::thread(
+	    [ss_width, ss_height](std::vector<u8> sshot_data) {
+		    std::string screen_path = fmt::format("%s/Screenshots/", fs::get_config_dir());
+
+		    if (!fs::create_dir(screen_path) && fs::g_tls_error != fs::error::exist)
+		    {
+			    LOG_ERROR(GENERAL, "Failed to create screenshot path \"%s\" : %s", screen_path, fs::g_tls_error);
+			    return;
+		    }
+
+		    std::time_t tt       = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+		    std::tm tm           = *std::localtime(&tt);
+		    std::string filename = fmt::format("%sScreenshot-%04d-%02d-%02d-%02d-%02d-%02d.png", screen_path, tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday, tm.tm_hour, tm.tm_min, tm.tm_sec);
```

Comment:
There's actually a method somewhere to do this. You could improve that one and reuse it here. (See rsx captures)